tests/rptest/tests/metrics_reporter_test.py

```diff
@@ -80,10 +80,13 @@ def assert_fields_are_the_same(metadata, field):
         # cluster uuid and create timestamp should stay the same across requests
         assert_fields_are_the_same(metadata, 'cluster_uuid')
         assert_fields_are_the_same(metadata, 'cluster_created_ts')
+        # Configuration should be the same across requests
+        assert_fields_are_the_same(metadata, 'has_kafka_gssapi')
         # get the last report
         last = metadata.pop()
         assert last['topic_count'] == total_topics
         assert last['partition_count'] == total_partitions
+        assert last['has_kafka_gssapi'] is False
```

Comment:
if we factor out the fetching of this metadata, then perhaps we could call that and have a stronger assertion in the kerberos tests where we can see the flag change rather than just exist?

no need to fix in this PR but maybe convert this comment into a follow up ticket and someone can pick it up.

Reply:
https://github.com/redpanda-data/redpanda/issues/8309